diy-omd.ino

```diff
@@ -85,6 +106,244 @@ void loop() {
+    timeMillis = 0;
+  }
+  // Serial.println(hallEffectCounter);
+  // delay(1);
+}
+
+void processHallSensor() {
+  const int hallSensorValue = analogRead(hallEffectSensorPin);
+
+  if (goingUp) {
+    if (hallSensorValue > HALL_THRESHOLD2) {
+      reachedThreshold2 = true;
+    } else if (hallSensorValue > HALL_THRESHOLD1) {
+      reachedThreshold1 = true;
+      reachedThreshold2 = false;
+    }
+
+    if (reachedThreshold1 && reachedThreshold2) {
+      revolutions++;
+      reachedThreshold1 = false;
+      reachedThreshold2 = false;
+      goingUp = false;
+    }
+  } else {
+    if (hallSensorValue < HALL_THRESHOLD1) {
+      goingUp = true;
+    }
+  }
+}
+
+void init() {
+  ESC.write(stopValue);
+  //Serial.println("init");
+  if (secondaryTimeMillis > 3000 && (MODE == OMD || MODE == BALANCE)) {
+    //Serial.println("init done");
+    motorState = GO;
+    secondaryTimeMillis = 0;
+  } else if (MODE == AXE_550_CALIBRATE) {
+    // Serial.println("AXE_550_calibrate init");
+    // Serial.println(secondaryTimeMillis);
+    digitalWrite(13, secondaryTimeMillis % 2000 > 1000 ? HIGH : LOW);
+
+    if (secondaryTimeMillis > 10000) {
+      motorState = GO;
+      secondaryTimeMillis = 0;
+    }
+  }
+}
+
+void go() {
+  digitalWrite(13, HIGH);
+  ESC.write(goValue);
+  if (secondaryTimeMillis > onSequence && MODE == OMD) {
+    //Serial.println("go done");
+    motorState = STOP;
+    secondaryTimeMillis = 0;
+  } else if (MODE == BALANCE) {
+    // do nothing: we just keep going in balance mode!
+  } else if (MODE == AXE_550_CALIBRATE) {
+    // Serial.println("AXE_550_calibrate go");
+    // Serial.println(secondaryTimeMillis % 1000);
+    digitalWrite(13, secondaryTimeMillis % 1000 > 500 ? HIGH : LOW);
+
+    if (secondaryTimeMillis > 10000) {
+      motorState = REVERSE;
+      secondaryTimeMillis = 0;
+    }
+  }
+}
+
+void processMode(const char* modeString) {
+  int mode = atoi(kModeToCode[modeString]);
+
+  Serial.println("Switching mode to");
+  switch (mode) {
+    case OMD: {
+      Serial.println("OMD");
+      MODE = OMD;
+      secondaryTimeMillis = 0;
+      break;
+    }
+
+    case BALANCE: {
+      Serial.println("BALANCE");
+      MODE = BALANCE;
+      break;
+    }
+
+    case AXE_550_CALIBRATE: {
+      Serial.println("CALIBRATE");
+      MODE = AXE_550_CALIBRATE;
+      break;
+    }
+
+    case OFF: {
+      Serial.println("OFF");
+      MODE = OFF;
+      break;
+    }
+  }
+
+  int tempGoValue = MODE == OMD ? OMD_goValue : (
+    MODE == BALANCE ? BALANCE_goValue : AXE_550_CALIBRATE_goValue
+  );
+
+  if (tempGoValue <= MAX_GO_VALUE) {
+    goValue = MODE == OMD ? OMD_goValue : (
+      MODE == BALANCE ? BALANCE_goValue : AXE_550_CALIBRATE_goValue
+    );
+  }
+}
+
+void processIncomingBTData() {
+  char appData;
+
+  if (btSerial.available() > 0) {
+    appData = btSerial.read();
+
+    if (appData == PARSE_START || start == true) {
+      inData += appData;  // save the data in string format
```

Comment:
`inData` could be a local variable here, which we pass directly by argument to `processJson`.